### ISSUE TITLE
`RET50*` ignores try/else blocks

### ISSUE THREAD

**Issue by korverdev** [astral-sh/ruff#8018]
This is probably harder to check for and much more uncommon, but RET505 could potentially apply to `try/else` blocks as well.

Example of the admittedly pretty bad code where this cropped up for me:
```python
try:
    run("docker --version", check=True, shell=True)
except CalledProcessError:
    return "podman"
else:
    return "docker"
```

What I'd expect this to be changed to:
```python
try:
    run("docker --version", check=True, shell=True)
except CalledProcessError:
    return "podman"
return "docker"
```

**Comment by tdulcet** [comment]
#970 says that [Pylint R1705](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/no-else-return.html) is implemented by `RET505`, but Pylint does supports else blocks on try statements, while this rule does not.

Also see #9625.

**Comment by charliermarsh** [comment]
Seems reasonable to support.

**Comment by TomMontagnon** [comment]
It doesn't manage `elif` statements well too.

This `elif` raise `RET505`, but is clearly not superflous.

```
def my_fonc():
    While True
        resp = input("y/n question : ")
        if resp == "y":
            foo()
            return True
        elif resp == "n":
            bar()
            return False
        else:
            print("try again")
```

**Comment by tdulcet** [comment]
> This `elif` raise `RET505`, but is clearly not superfluous.

Both the `elif` and `else` in your example look superfluous to me:
```py
def my_fonc():
    while True:
        resp = input("y/n question : ")
        if resp == "y":
            foo()
            return True

        if resp == "n":
            bar()
            return False

        print("try again")
```

**Comment by TomMontagnon** [comment]
Yup, I hadn't seen things that way.... you are perfectly right, thanks !

**Comment by MichaReiser** [comment]
@seroperson created a great PR adding support for this, but we realized that this is more tricky with try..except because there are multiple preceding branches:

Here's a copy of my comment on the PR:

Codex found a good reason why it's worth considering whether this should be its own rule:

```py
def foo():
    try:
        foo()
    except ValueError:
        return 1
    except TypeError:
        raise RuntimeError
    else: # noqa RET505
        return 2

def f():
    try:
        foo()
    except TypeError:
        raise RuntimeError
    except ValueError:
        return 1
    else: # noqa RET506
        return 2
```

The two try blocks are identical except that the order of the except handlers is flipped. 

* Example 1: The first `except` handler uses `return`, the second handler uses `raise`
* Example 2: The first `except` handler uses `raise`, the second uses `return`

The important bit is that it's the last except handler that defines the error code. This seems "weird" if a try statement has both except handlers that use `raise` and `return`. It's then unclear which error code is correct.

Arguably, this is an inherent downside of all `RET` rules, and it's unclear how these rules will evolve if we have a more sophisticated analysis for detecting whether a body always returns, e.g., what about:

```py
def foo(arg: bool):
	if a:
		if arg:
			return True
		else if:
			raise ValueError("bla")
	else:
		print("test")
```

The `else` branch is unnecessary because the `a==True` branch always terminates. 


But given that this is worse for `try`, I don't feel comfortable extending the existing rules. Introducing a new rule is an option, but we have to carefully think about how it overlaps with the existing `RET` rules and how we envision the rules in the future (deprecate `RET5XX` and merge them all into a new rule`). 

What's unclear to me is the motivation for having separate rule codes to begin with. Is it common for users to enable only some `RET50X` rules, not others? If so, why? If not, then the best path forward would be to have a single `superfluous-else` rule that covers all `RET50x` rules.
